File: nhglib-core/src/io/github/voidzombie/nhglib/data/models/serialization/Vector3Json.java
package io.github.voidzombie.nhglib.data.models.serialization;

import com.badlogic.gdx.math.Vector3;
import com.badlogic.gdx.utils.JsonValue;
import io.github.voidzombie.nhglib.interfaces.JsonParseable;

/**
 * Created by Fausto Napoli on 11/06/2017.
 */
public class Vector3Json implements JsonParseable<Vector3> {
    private Vector3 vec;

    @Override
    public void parse(JsonValue jsonValue) {
        vec = new Vector3(
                jsonValue.getFloat("x"),
                jsonValue.getFloat("y"),
                jsonValue.getFloat("z"));
    }

    @Override
    public Vector3 get() {
        return vec;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/Vector3Json.kt
package io.github.movementspeed.nhglib.data.models.serialization

import com.badlogic.gdx.math.Vector3
import com.badlogic.gdx.utils.JsonValue
import io.github.movementspeed.nhglib.interfaces.JsonParseable

/**
 * Created by Fausto Napoli on 11/06/2017.
 */
class Vector3Json : JsonParseable<Vector3> {
    private var vec: Vector3? = null

    override fun parse(jsonValue: JsonValue) {
        vec = Vector3(
                jsonValue.getFloat("x"),
                jsonValue.getFloat("y"),
                jsonValue.getFloat("z"))
    }

    override fun get(): Vector3? {
        return vec
    }
}
